core_lib/src/graphics/bitmap/bitmapimage.cpp

```diff
@@ -85,6 +85,21 @@ BitmapImage* BitmapImage::clone()
     return new BitmapImage(*this);
 }
 
+void BitmapImage::loadFile()
+{
+    mImage = std::make_shared<QImage>(fileName());
```

Comment:
Maybe you should add a check here to make sure that `mImage == nullptr || !isModified()`. If this is accidentally called from somewhere else when this is not a nullptr, then changes to the image could be overwritten which is almost certainly not the desired behavior.

Reply:
Good advice. will do that.